test/python/circuit/test_library.py

```diff
@@ -50,3 +55,108 @@ def test_inner_product(self):
         expected.cz(1, 4)
         expected.cz(2, 5)
         self.assertEqual(circuit, expected)
+
+
+@ddt
+class TestBasisChanges(QiskitTestCase):
+    """Test the basis changes."""
+
+    def to_matrix(self, circuit):
+        """Get the unitary matrix from simulating the circuit with the unitary simulator."""
+        backend = BasicAer.get_backend('unitary_simulator')
+        return execute(circuit, backend).result().get_unitary()
+
+    def assertQFTIsCorrect(self, qft, num_qubits=None, inverse=False, prepend_swaps=False):
+        """Assert that the QFT circuit produces the correct matrix.
+
+        Can be provided with an explicit number of qubits, if None is provided the number
+        of qubits is set to ``qft.num_qubits``.
```

Comment:
`num_qubits` is not different to `qft.num_qubits` anywhere, and this function kind of expects them to be equal. So I think this `num_qubits` kwarg can be removed.

Reply:
This implicitly checks that the setter `qft.num_qubits` is working. If it wouldn't be working and e.g. the number of qubits stayed the same, the test would still get the old number from the `qft` instance. This way we can assert that we test against a matrix which has the size we want